### ISSUE TITLE
New `IrrConlon` method doesn't work for all groups

### ISSUE THREAD

**Issue by apfelkuchen6** [gap-system/gap#4940]
The new `IrrConlon` implementation introduced in 44318c4d8f255e143c99334a7844ded1847fbf68 yields incorrect results for some groups.

Let's validate it against `IrrBaumClausen` and `IrrDixonSchneider` with a p-group (for which all algorithms should compute all irreducible characters):

### Observed behaviour
```gap
gap> G := Group([ (5,9)(10,14)(11,12)(15,16), (2,3)(4,7)(5,10)(9,14)(11,12)(15,16), (2,4)(3,5)(6,12)(7,9)(10,14)(11,13), (1,2)(3,6)(4,8)(5,11)(7,13)(9,12)(10,15)(14,16) ]);;
gap> Order(G);
1024
gap> c := Set(IrrConlon(G));;
gap> bc := Set(IrrBaumClausen(G));;
gap> ds := Set(IrrDixonSchneider(G));;
gap> bc =ds;
true
gap> c = bc;
false
```

The character table computed with `IrrConlon` can't even be `Display`ed (it looks like it chokes when trying to compute the centralizer orders because of incompatible information)
```gap
gap>Display(CharacterTable(G));
CT1

Error, List Element: <list>[5] must have an assigned value in
  cen[prime][i] := Number( fak, function ( x )
        return x = prime;
    end ); at /nix/store/id9bca8bz8qy1zwi2zg5j4drsa9i92j6-gap-4.12pre-2022-07-10/share/gap/build-dir/lib/ctbl.gi:5151 called from 
record.Display( tbl, record ); at /nix/store/id9bca8bz8qy1zwi2zg5j4drsa9i92j6-gap-4.12pre-2022-07-10/share/gap/build-dir/lib/ctbl.gi:5334 called from
Display( tbl, CharacterTableDisplayDefaults.Global ); at /nix/store/id9bca8bz8qy1zwi2zg5j4drsa9i92j6-gap-4.12pre-2022-07-10/share/gap/build-dir/lib/ctbl.gi:5318 called from
cen[prime][i] := Number( fak, function ( x )
        return x = prime;
    end ); at /nix/store/id9bca8bz8qy1zwi2zg5j4drsa9i92j6-gap-4.12pre-2022-07-10/share/gap/build-dir/lib/ctbl.gi:5151 called from
record.Display( tbl, record ); at /nix/store/id9bca8bz8qy1zwi2zg5j4drsa9i92j6-gap-4.12pre-2022-07-10/share/gap/build-dir/lib/ctbl.gi:5334 called from
Display( tbl, CharacterTableDisplayDefaults.Global ); at /nix/store/id9bca8bz8qy1zwi2zg5j4drsa9i92j6-gap-4.12pre-2022-07-10/share/gap/build-dir/lib/ctbl.gi:5318 called from
...  at *errin*:1
type 'quit;' to quit to outer loop
```

### Expected behaviour
```gap
gap> c = bc;
true
```

### Copy and paste GAP banner (to tell us about your setup)
rev 0deabeb8f04dd60727534cbdcb5b94aa4766a0b9,
```
 ┌───────┐   GAP 4.12dev built on 2022-07-13 10:46:58+0000
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64-kv8
 Configuration:  gmp 6.2.1, GASMAN, readline
 Loading the library and packages ...
 Packages:   AClib 1.3.2, Alnuth 3.1.2, AtlasRep 2.1.0, AutoDoc 2020.08.11, 
             AutPGrp 1.10.2, Browse 1.8.13, CaratInterface 2.3.3, CRISP 1.4.5, 
             Cryst 4.1.24, CrystCat 1.1.9, CTblLib 1.3.3, FactInt 1.6.3, 
             FGA 1.4.0, Forms 1.2.6, GAPDoc 1.6.5, genss 1.6.6, 
             InduceReduce 1.0, IO 4.7.2, IRREDSOL 1.4.3, LAGUNA 3.9.3, 
             orb 4.8.4, Polenta 1.3.9, Polycyclic 2.16, PrimGrp 3.4.1, 
             RadiRoot 2.8, recog 1.3.2, ResClasses 4.7.2, SmallGrp 1.4.2, 
             Sophus 1.24, SpinSym 1.5.2, TomLib 1.2.9, TransGrp 3.3, utils 0.72
             
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'

```

**Comment by fingolfin** [comment]
@apfelkuchen6 thank you for taking the time to report this regression! I hope @frankluebeck will have time to look into it.

**Comment by frankluebeck** [comment]
Many thanks to @apfelkuchen6 for this example! The example helped to identify and fix this (embarrasing) problem. 

As I already did on ealier occasions I mixed up the meaning of "right coset representatives" (vs. left coset representatives).
In `IrrConlon` many characters are induced from normal subgroups and also if not, right coset reps are often a good "approximation" to left coset reps. That must be the reason why I did not notice the bug when testing for about 150000 supersolvable groups of order up to 1023, if  `IrrBaumClausen` and `IrrConlon` yield the same result.
